File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
    var max int
    for i:=0;i<len(candies);i++{
        if candies[i]>max {
            max=candies[i]
        }
    }
    var res []bool
    for i:=0;i<len(candies);i++{
        if (candies[i]+extraCandies >= max){
            res=append(res,true)
        } else {
            res=append(res,false)
        }
    }
    return res
}

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.php
class Solution {

    /**
     * @param Integer[] $candies
     * @param Integer $extraCandies
     * @return Boolean[]
     */
    function kidsWithCandies($candies, $extraCandies) {
        $max = max($candies);
        $res = [];
        
        foreach($candies as $candy){
            if($candy + $extraCandies >= $max){
                $res[] = true;
                continue;
            }
            
            $res[] = false;
        }
        
        return $res;
    }
}